app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -1,71 +1,46 @@
 import { TextInput } from '@18f/identity-components';
-import { request } from '@18f/identity-request';
-import { useState, useCallback, ChangeEvent, useRef, useEffect } from 'react';
+import { useState, useRef, useEffect, RefObject, useCallback } from 'react';
 import { useI18n } from '@18f/identity-react-i18n';
 import ValidatedField from '@18f/identity-validated-field/validated-field';
 import SpinnerButton, { SpinnerButtonRefHandle } from '@18f/identity-spinner-button/spinner-button';
 import type { RegisterFieldCallback } from '@18f/identity-form-steps';
-import useSWR from 'swr';
-
-interface Location {
-  street_address: string;
-  city: string;
-  state: string;
-  zip_code: string;
-  address: string;
-}
 
 interface AddressSearchProps {
-  onAddressFound?: (location: Location) => void;
+  onSearch?: (
+    event: MouseEvent,
+    textInput: string,
+    fieldValidationRef: RefObject<HTMLFormElement> | undefined,
+  ) => void;
   registerField?: RegisterFieldCallback;
+  loading?: boolean;
 }
 
 export const ADDRESS_SEARCH_URL = '/api/addresses';
 
-function requestAddressCandidates(unvalidatedAddressInput: string): Promise<Location[]> {
-  return request<Location[]>(ADDRESS_SEARCH_URL, {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' },
-    json: { address: unvalidatedAddressInput },
-  });
-}
-
 function AddressSearch({
-  onAddressFound = () => {},
   registerField = () => undefined,
+  onSearch = () => {},
+  loading = false,
 }: AddressSearchProps) {
-  const validatedFieldRef = useRef<HTMLFormElement | null>(null);
-  const [unvalidatedAddressInput, setUnvalidatedAddressInput] = useState('');
-  const [addressQuery, setAddressQuery] = useState('');
   const { t } = useI18n();
-  const { data: addressCandidates } = useSWR([ADDRESS_SEARCH_URL, addressQuery], () =>
-    addressQuery ? requestAddressCandidates(unvalidatedAddressInput) : null,
-  );
-  const ref = useRef<SpinnerButtonRefHandle>(null);
+  const spinnerButtonRef = useRef<SpinnerButtonRefHandle>(null);
+  const validatedFieldRef = useRef<HTMLFormElement>(null);
+  const [textInput, setTextInput] = useState('');
 
-  useEffect(() => {
-    if (addressCandidates) {
-      const bestMatchedAddress = addressCandidates?.[0];
-      const validity = bestMatchedAddress ? '' : t('in_person_proofing.body.location.inline_error');
+  const onTextInputChange = useCallback((event) => {
+    const target = event.target as HTMLInputElement;
```

Comment:
The better solution for here is to set the parameter type instead of casting. e.g. using `React.FormEvent<HTMLInputElement>`.